src/main/OP.ts

```diff
@@ -44,21 +43,26 @@ export class OP {
     return this._verifyAuthRequestOpts;
   }
 
+  // TODO SK Can you please put some documentation on it?
   public async postAuthenticationResponse(authenticationResponse: AuthenticationResponseWithJWT): Promise<Response> {
     return postAuthenticationResponse(authenticationResponse.payload.aud, authenticationResponse);
   }
 
   public async verifyAuthenticationRequest(
-    requestJwtorUri: string,
+    requestJwtOrUri: string,
     opts?: { nonce?: string; verification?: InternalVerification | ExternalVerification }
   ): Promise<VerifiedAuthenticationRequestWithJWT> {
-    const jwt = requestJwtorUri.startsWith('ey') ? requestJwtorUri : (await parseAndResolveUri(requestJwtorUri)).jwt;
-    const verifiedJwt = AuthenticationRequest.verifyJWT(jwt, this.newVerifyAuthenticationRequestOpts(opts));
+    const jwt = requestJwtOrUri.startsWith('ey') ? requestJwtOrUri : (await parseAndResolveUri(requestJwtOrUri)).jwt;
+    const verifiedJwt = AuthenticationRequest.verifyJWT(
+      jwt,
+      this.newVerifyAuthenticationRequestOpts(opts),
+      this._authResponseOpts.linkedDomainValidationMode
```

Comment:
Hmzz, this doesn't look like it is inline with other areas. Probably the opts needs to be extracted into an interface and the domain linkage should be added there

Reply:
yes, changed